pkg/common/utils/utils.go

```diff
@@ -154,15 +163,26 @@ func GetApplicationIDFromPod(pod *v1.Pod) string {
 		}
 	}
 
-	// Application ID can be defined in annotation
-	appID := GetPodAnnotationValue(pod, constants.AnnotationApplicationID)
-	if appID == "" {
-		// Application ID can be defined in label
-		appID = GetPodLabelValue(pod, constants.LabelApplicationID)
+	// Application ID can be defined in multiple places
+	// The application ID is determined by the following order.
+	// 1. Label: constants.CanonicalLabelApplicationID
+	// 2. Label: constants.LabelApplicationID
+	// 3. Label: constants.SparkLabelAppID
+	// 4. Annotation: constants.AnnotationApplicationID
+	labelKeys := []string{
+		constants.CanonicalLabelApplicationID,
+		constants.LabelApplicationID,
+		constants.SparkLabelAppID,
+	}
+	appID := ""
+	for _, label := range labelKeys {
+		appID = GetPodLabelValue(pod, label)
+		if appID != "" {
+			break
+		}
 	}
 	if appID == "" {
-		// Spark can also define application ID
-		appID = GetPodLabelValue(pod, constants.SparkLabelAppID)
+		appID = GetPodAnnotationValue(pod, constants.AnnotationApplicationID)
 	}
```

Comment:
The order to get 'app-id' from pod:

Before this PR:
1. Annotation: constants.AnnotationApplicationID
2. Label: constants.LabelApplicationID
3. Label: constants.SparkLabelAppID

After this PR
1. **Label: constants.CanonicalLabelApplicationID (New)**
2. Label: constants.LabelApplicationID
3. Label: constants.SparkLabelAppID
4. **Annotation: constants.AnnotationApplicationID(Move to the last, label first)**